lib/smart_proxy_dns_powerdns/dns_powerdns_main.rb

```diff
@@ -74,6 +79,18 @@ def domain_id
     end
 
     private
+    def domain
```

Comment:
I wonder if I should refactor domain_id to return a tuple of ID and name and store it as variables to avoid doing so many queries. @domcleal agreed?